backend/app/dependencies.py

```diff
@@ -1,31 +1,30 @@
 from typing import Annotated
 
-from fastapi import Depends
+from fastapi import Depends, Request
 from fastapi.security import OAuth2PasswordBearer
 from pydantic import SecretStr
 
-from app.auth import AuthException
-from app.auth.supabase import SupabaseAuthService
-from app.models import AuthToken, CredentialsError, settings
+from app.auth import AuthException, UserResponse
+from app.models import AuthToken, CredentialsError
 
 oauth2_scheme = OAuth2PasswordBearer(tokenUrl="login")
 
 
 async def check_user_authentication(
-    access_token: Annotated[SecretStr, Depends(oauth2_scheme)],
+    access_token: Annotated[SecretStr, Depends(oauth2_scheme)], request: Request
 ) -> None:
     """
     Check if `access_token` corresponds to an authenticated `User`.
 
     A `CredentialsError` exception is raised if the authentication fails.
     """
-    supabase_auth_service = SupabaseAuthService(settings=settings)
 
     try:
-        supabase_auth_service.get_user(
+        user_response: UserResponse = request.app.state.auth_service.get_user(
             authentication_token=AuthToken(access_token=access_token)
         )
     except AuthException as exception:
-        raise CredentialsError(
-            detail=f"Could not validate credentials. Encountered exception: {exception.message}"
-        ) from exception
+        raise CredentialsError from exception
+
+    if not user_response.succeeded:
```

Comment:
Code smell: if a falsey turns into an exception then it should probably be an exception originally.

Reply:
True. The false-value check has been removed.